File: leetcode/1137.py
from functools import cache

class Solution:
    @cache
    def tribonacci(self, n: int) -> int:
        match n:
            case 0:
                return 0
            case 1 | 2:
                return 1
            case _:
                return self.tribonacci(n-1) + self.tribonacci(n-2) + self.tribonacci(n-3)

File: leetcode/1137.rs
impl Solution {
    pub fn tribonacci(n: i32) -> i32 {
        fn trib(n: i32, a: i32, b: i32, c: i32) -> i32 {
            match n {
                0 => a,
                1 => b,
                2 => c,
                _ => trib(n - 1, b, c, a + b + c),
            }
        }        
        
        trib(n, 0, 1, 1)
    }
}
